src/utils/api.js

```diff
@@ -1,20 +1,21 @@
-export const createEnvironment = async (authContext) => {
+const apiCall = async (authContext, endpoint) => {
     const user = await authContext.getCurrentUser();
     const jwtToken = user.signInUserSession.idToken.jwtToken;
-    const res = await fetch(`/connect?id_token=${jwtToken}`);
-    if (res.status === 200) return `https://${await res.text()}?token=${jwtToken}`
+    const res = await fetch(`/${endpoint}?id_token=${jwtToken}`);
+
+    if (res.status === 200){
+        switch (endpoint) {
```

Comment:
Not a fan of trying to have a single method multiplex, rather than having separate methods. It's a bad anti-pattern to overload methods. However since SonarCloud isn't complaining, I'll let it pass.